src/Resources/views/CRUD/Association/edit_many_script.html.twig

```diff
@@ -361,18 +361,36 @@ This code manages the many-to-[one|many] association field popup
                 if ('application/json' === xhr.getResponseHeader('Content-Type')) {
                     var jsonContent = JSON.parse(xhr.responseText);
 
-                    if (jsonContent.message) {
+                    if (jsonContent.title) {
                         content += '<div class="alert alert-danger alert-dismissable">'
                             + '<button type="button" class="close" data-dismiss="alert" aria-hidden="true">×</button>'
-                            + jsonContent.message
+                            + jsonContent.title
                             + '</div>';
                     }
 
-                    if (jsonContent.errors) {
-                        for (error of jsonContent.errors) {
+                    // Clean up all previous errors that we have attached to the form, in case of second request.
+                    $(form).find('.xhr-violation-error').removeClass('has-error');
+                    $(form).find('.xhr-violation-error-block').remove();
+
+                    if (jsonContent.violations) {
+                        for (violation of jsonContent.violations) {
+                            // Try to find and attach the error to the form field, if not show message on top.
+                            if (violation.hasOwnProperty('propertyPath')) {
+                                const field = $(form).find(`[name='${violation.propertyPath}']`);
+                                if (field.length) {
+                                    field.closest('.form-group').addClass('has-error xhr-violation-error');
+                                    field.parent().append(`<div class="help-block sonata-ba-field-error-messages xhr-violation-error-block">
```

Comment:
This won't work if one field triggers 2 validation errors. IMO this code should be written in plain JS, and you can use some (or all) the logic written here: https://github.com/sonata-project/SonataPageBundle/blob/4.x/assets/js/composer.js#L255-L300

It should be quite similar.